spec/services/billable_metrics/aggregations/sum_service_spec.rb

```diff
@@ -225,23 +267,80 @@
   end
 
   context 'when event is given' do
+    let(:old_events) { nil }
+    let(:latest_events) { nil }
     let(:pay_in_advance_event) do
       create(
         :event,
         code: billable_metric.code,
         customer:,
         subscription:,
-        timestamp: Time.zone.now - 1.day,
+        timestamp: to_datetime - 2.days,
         properties:,
       )
     end
 
-    let(:properties) { { total_count: 12 } }
+    let(:properties) { { total_count: 10 } }
 
     it 'assigns a pay_in_advance aggregation' do
       result = sum_service.aggregate(from_datetime:, to_datetime:)
 
-      expect(result.pay_in_advance_aggregation).to eq(12)
+      expect(result.pay_in_advance_aggregation).to eq(10)
+    end
+
+    context 'when current period aggregation is greater than period maximum' do
+      let(:latest_events) do
+        create(
+          :event,
+          code: billable_metric.code,
+          customer:,
+          subscription:,
+          timestamp: to_datetime - 3.days,
+          properties: {
+            total_count: -6,
+          },
+          metadata: {
+            current_aggregation: '4',
+            max_aggregation: '10',
+          }
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)